backends/smt2/smtio.py

```diff
@@ -232,17 +232,21 @@ def setup(self):
             if self.logic_uf: self.logic += "UF"
             if self.logic_bv: self.logic += "BV"
             if self.logic_dt: self.logic = "ALL"
+            if self.solver == "yices" and self.forall: self.logic = "BV"
 
         self.setup_done = True
 
-        for stmt in self.info_stmts:
-            self.write(stmt)
+        if self.forall and self.solver == "yices":
+            self.write("(set-option :yices-ef-max-iters 1000000000)")
```

Comment:
Yices has a default exists-forall mode iteration limit of 100.  This is not nearly enough to solve pretty much any problem I've ever generated.  I set the iteration limit to a billion here because we don't care how many iterations the solver can do.  At most we care about limiting the time the solver can take, but that is a separate issue and is addressed in another forthcoming PR.